akka-remote/src/main/scala/akka/remote/artery/compress/InboundCompressions.scala

```diff
@@ -213,6 +185,8 @@ private[remote] abstract class InboundCompression[T >: Null](
       finally scheduleNextTableAdvertisement()
   }
 
+  private[this] var advertisementInProgress = false
```

Comment:
The AtomicReference idea is to be challanged though, as perhaps we can get away without it, if we make ownership clear